backend/danswer/search/search_nlp_models.py

```diff
@@ -103,28 +105,60 @@ def __init__(
         model_server_url = build_model_server_url(server_host, server_port)
         self.embed_server_endpoint = f"{model_server_url}/encoder/bi-encoder-embed"
 
-    def encode(self, texts: list[str], text_type: EmbedTextType) -> list[list[float]]:
+    def encode(
+        self,
+        chunk_texts: list[str],
+        text_type: EmbedTextType,
+        batch_size: int = BATCH_SIZE_ENCODE_CHUNKS,
+    ) -> list[list[float]]:
         if text_type == EmbedTextType.QUERY and self.query_prefix:
-            prefixed_texts = [self.query_prefix + text for text in texts]
+            prefixed_texts = [self.query_prefix + text for text in chunk_texts]
         elif text_type == EmbedTextType.PASSAGE and self.passage_prefix:
-            prefixed_texts = [self.passage_prefix + text for text in texts]
+            prefixed_texts = [self.passage_prefix + text for text in chunk_texts]
         else:
-            prefixed_texts = texts
-
-        embed_request = EmbedRequest(
-            model_name=self.model_name,
-            texts=prefixed_texts,
-            max_context_length=self.max_seq_length,
-            normalize_embeddings=self.normalize,
-            api_key=self.api_key,
-            provider_type=self.provider_type,
-            text_type=text_type,
-        )
+            prefixed_texts = chunk_texts
+
```

Comment:
if a cloud provider is given, we want to do all the chunks at once and let the cloud provider handle it